.buildkite/filebeat/filebeat-pipeline.yml

```diff
@@ -1,5 +1,157 @@
 # yaml-language-server: $schema=https://raw.githubusercontent.com/buildkite/pipeline-schema/main/schema.json
 
+env:
+  SETUP_GVM_VERSION: "v0.5.1"
```

Comment:
If this version will be used elsewhere, what do you think using the `pre-command` hook instead?

Reply:
Generally speaking, I have never seen that env vars would be defined in pre-command hooks, except for the case, when those are exported to env in Buildkite pipelines. So I'd assume it's general pattern to store those closer to the pipeline, since it's more readable and easier to change without risk of interfering other pipelines. And another thing, since there several people working on beats migration, we aim to work iteratively - migrate pipelines so each of them works to avoid any concurrent issues, and after that make optimizations.

Reply:
In this case, `gvm` is an internal tool to install the give Go version in `.go-version`. IMO, this is not needed in each pipeline.

My concern was more about usability; centralising the variables in one place helps consumers not to need to care about them but to use them. https://github.com/elastic/kibana/blob/main/.buildkite/scripts/common/env.sh is how the Kibana team manages the different environment variables; the `pre-command` hook calls that script.

>> Migrate pipelines so each of them works to avoid any concurrent issues, and after that, make optimisations.

Sure thing